src/xrpld/app/tx/detail/applySteps.cpp

```diff
@@ -294,15 +304,22 @@ preflight(
     ApplyFlags flags,
     beast::Journal j)
 {
-    PreflightContext const pfctx(app, tx, rules, flags, j);
+    bool const isDelegated =
+        rules.enabled(featureAccountPermission) && tx[~sfOnBehalfOf];
+    AccountID const account = isDelegated ? *tx[~sfOnBehalfOf] : tx[sfAccount];
+    PreflightContext const pfctx(app, tx, rules, flags, account, j);
     try
     {
-        return {pfctx, invoke_preflight(pfctx)};
+        // if AccountPermission is not enabled, do not use OnBehalfOf field.
+        if (!rules.enabled(featureAccountPermission) && tx[~sfOnBehalfOf])
```

Comment:
so what's the behavior when someone submits a tx with sfOnBehalfOf but not enabled the amendment? what error code does it return. To me, i feel we should not be throwing a runtime error, because it's a completely acceptable input. pls consider performing this check in the `Transactor::preflight1` instead and return an code like `temDISABLED`